src/extension.ts

```diff
@@ -247,6 +254,56 @@ class DSSExtension {
         });
     }
     
+    async openWikiArticleFile(item: WikiArticleTreeView) {
+        item.dssObject = await getWikiArticle(item.dssObject.article.projectKey, item.dssObject.article.id);
+        const filePath = await this.fsManager.saveInFS(FileDetails.fromWikiArticle(item.dssObject));
+        await this.openTextDocumentSafely(filePath, item);    
+    }
+    
+    async addWikiArticle(parentItem: WikiFolderTreeView | WikiArticleTreeView) {
+        const filename = await vscode.window.showInputBox();
+        if (filename) {
+            try {
+                if (parentItem instanceof WikiFolderTreeView) {
+                    await createWikiArticle(parentItem.parent.dssObject.projectKey, filename)
+                }
+                else {
+                    await createWikiArticle(parentItem.dssObject.article.projectKey, filename, parentItem.dssObject.article.id);
+                }
+                vscode.commands.executeCommand("dssProjects.refreshEntry");
+            } catch {
+                vscode.window.showErrorMessage(`Can not create wiki article with name ${filename}`);
+            }
+        }
+    }
+    
+    async deleteWikiArticle(item: WikiArticleTreeView) {
+        await deleteWikiArticle(item.dssObject);
+        vscode.commands.executeCommand("dssProjects.refreshEntry");
+        vscode.window.showInformationMessage("Wiki article \"" + item.dssObject.article.name + "\" deleted successfully");
+    }
+    
+    openInDSS(item: ProjectsFolderTreeView | RecipesFolderTreeView | WebAppsFolderTreeView | WikiFolderTreeView | WikiArticleTreeView | RecipeFileTreeView | WebAppFolderTreeView | RootPluginFolderTreeView) {
```

Comment:
To simplify `openInDss` I would suggest to use inheritance instead of if/else. We could create a abstract class (something like `OpenableInDSS` ?) which will have a method `urlInDSS()`. All the accepted classes as `openInDSS` parameters would have to inherit `OpenableInDSS` and override `urlInDSS()`. So `openInDss` would like this:
```javascript
openInDSS(item: OpenableInDSS) {
    vscode.env.openExternal(item.urlInDSS());
}
```

Reply:
Great idea (again)! :)
I preferred to use an interface that I though was more appropriate to me than using inheritance.
Let me know if you're okay with this.

Reply:
It's perfect 👌